pyhive/presto.py

```diff
@@ -104,7 +108,10 @@ def __init__(self, host, port='8080', username=None, catalog='hive', schema='def
         if protocol not in ('http', 'https'):
             raise ValueError("Protocol must be http/https, was {!r}".format(protocol))
         self._protocol = protocol
-
+        self._auth = None
```

Comment:
nitpick: move to else branch of if password, so we're not setting it and mutating it